hack/generated/controllers/generic_controller.go

```diff
@@ -194,7 +193,9 @@ func (gr *GenericReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 	// https://github.com/kubernetes/community/blob/master/contributors/devel/sig-api-machinery/controllers.md, which says:
 	// Never mutate original objects! Caches are shared across controllers, this means that if you mutate your "copy"
 	// (actually a reference or shallow copy) of an object, you'll mess up other controllers (not just your own).
-	obj = obj.DeepCopyObject()
+	obj = obj.DeepCopyObject().(client.Object)
+
+	gr.Log.V(0).Info("Reconcile invoked", "kind", fmt.Sprintf("%T", obj))
```

Comment:
minor: Is there a cleaner way to log the type being reconciled in the `logr.Logger` interface (since it doesn't seem to support format specifiers?)

Reply:
The best suggestion I have is to break out an intention revealing local:

```suggestion
    objKind := fmt.Sprintf("%T", obj)
	gr.Log.V(0).Info("Reconcile invoked", "kind", objKind)
```

Another way would be to introduce a function on each of our generated types that generates a description to use here, but that's a lot of overhead.